src/rust/organization-management/src/server.rs

```diff
@@ -0,0 +1,299 @@
+
+        transaction.commit().await?;
+
+        Ok(CreateOrganizationResponse { organization_id })
+    }
+
+    async fn create_user(
+        &self,
+        request: CreateUserRequest,
+    ) -> Result<CreateUserResponse, OrganizationManagementServiceError> {
+        let user_id = sqlx::types::Uuid::from_u128(Uuid::new_v4().as_u128());
+
+        let CreateUserRequest {
+            organization_id, // we need to do a lookup here
+            name,
+            email,
+            password,
+        } = request;
+
+        let password_hasher = argon2::Argon2::new(
+            argon2::Algorithm::Argon2i,
+            argon2::Version::V0x13,
+            argon2::Params::new(102400, 2, 8, None)?,
+        );
+
+        let password = password_hasher
+            .hash_password(&password, &SaltString::generate(OsRng))?
+            .serialize();
+
+        sqlx::query!(
+            r"
+             INSERT INTO users (
+                user_id,
+                organization_id,
+                username,
+                email,
+                password,
+                is_admin,
+                should_reset_password
+            )
+             VALUES ( $1, $2, $3, $4, $5, $6, $7);
+        ",
+            user_id,
+            organization_id,
+            name,
+            email,
+            password.as_str(),
+            false,
+            true,
+        )
+        .execute(&self.pool)
+        .await
+        .map_err(OrganizationManagementServiceError::from)?;
+
+        Ok(CreateUserResponse { user_id })
+    }
+}
+
+#[tonic::async_trait]
+impl OrganizationManagementService for OrganizationManagement {
+    async fn create_organization(
+        &self,
+        request: Request<CreateOrganizationRequestProto>,
+    ) -> Result<Response<CreateOrganizationResponseProto>, Status> {
+        let request: CreateOrganizationRequestProto = request.into_inner();
+        let request = CreateOrganizationRequest::try_from(request)
+            .map_err(OrganizationManagementServiceError::from)?;
+
+        let response = self.create_organization(request).await?;
+        let response: CreateOrganizationResponseProto = response.into();
+        Ok(Response::new(response))
+    }
+
+    async fn create_user(
+        &self,
+        request: Request<CreateUserRequestProto>,
+    ) -> Result<Response<CreateUserResponseProto>, Status> {
+        let request: CreateUserRequestProto = request.into_inner();
+        let request = CreateUserRequest::try_from(request)
+            .map_err(OrganizationManagementServiceError::from)?;
+
+        let response = self.create_user(request).await?;
+        let response: CreateUserResponseProto = response.into();
+        Ok(Response::new(response))
+    }
+}
+
+#[tracing::instrument(err)]
+async fn create_db_connection() -> Result<Pool<Postgres>, sqlx::Error> {
+    let url = "postgres://postgres@localhost?db_name=postgres&user=postgres&password=postgres";
+
+    println!("databse url {}", url);
```

Comment:
We shouldn't have any `println` calls; we're using the tracing infrastructure for this kind of stuff.